mirage/awa_mirage.ml

```diff
@@ -312,16 +312,21 @@ module Make (F : Mirage_flow.S) (T : Mirage_time.S) (M : Mirage_clock.MCLOCK) =
         let t = { t with channels = c :: t.channels } in
         nexus t fd server input_buffer (List.append pending_promises [ Lwt_mvar.take t.nexus_mbox ])
 
-  let spawn_server server msgs fd exec_callback =
+  let spawn_server ?stop server msgs fd exec_callback =
     let t = { exec_callback;
               channels = [];
               nexus_mbox = Lwt_mvar.create_empty ()
             }
     in
+    let switched_off =
+      let thread, u = Lwt.wait () in
+      Lwt_switch.add_hook stop (fun () ->
```

Comment:
this is a bit troublesome, from the documentation: `If [switch] contains an switch that is already off then {!Off} is raised.`

Should we instead use `add_hook_or_exec` or catch the `Off` exception?

Reply:
I decided to use the `add_hook_or_exec` 👍